File: java-project-euler-100/src/main/java/net/talaat/harb/projecteuler100/P001.java
package net.talaat.harb.projecteuler100;

/**
 * @author mharb
 *
 */
public class P001 implements Runnable {

	/**
	 * Problem solution entry point
	 * @param args
	 */
	public static void main(String[] args) {
		new P001().run();
	}

	@Override
	public void run() {
		int last = 1000;
		int sum = 0;

		for (int i = 1; i < last; i++) {
			if ((i % 3 == 0) || (i % 5 == 0)) {
				sum = sum + i;
			}
		}
		System.out.println("p001: " + sum);
	}

}

File: go-project-euler-100/p001.go
package main

/*P001Solve solution for p001*/
func P001Solve() int {
	sum := 0
	last := 1000
	for i := 1; i < last; i++ {
		if (i%3 == 0) || (i%5 == 0) {
			sum += i
		}
	}
	return sum
}
